### PR TITLE
makederiv

### PR DESCRIPTION
**Pull request by aeschweik**
Combines make scripts (makebroadcast, dvd, youtube, mp3, podcast, proves, resourcespace) into one derivative-creation script for streamlined code and editing. Plus a regression test to make sure tinkering works. I am sorry this got so big before I made a pull request! (New here)

### CODE REVIEW

File: makedvd [hunk 1]
```diff
@@ -1,172 +1,17 @@
...
-    echo "  -n (dry-run mode, show the commands that would be run but don't do anything)"
-    echo "  -e emailaddress ( send an email about the delivery, only valid if -d is used )"
-    echo "  -E emailaddress ( send an email about process outcome )"
-    echo "  -h ( display this help )"
     echo
-    exit
+    _report -d "makedvd is an alias for the updated function makederiv (options below)."
+    _report -d "You can create a DVD with makederiv by typing: 'makederiv -T dvd [options] [input]'."
+    _report -d "Or continue to use makedvd using the valid options below."
+    . "${SCRIPTDIR}/makederiv" -h
```
Comment (dericed) [inline makedvd]: do you mean
```
"${SCRIPTDIR}/makederiv" -h
```

File: mmtest [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+md5 -q "$MP3_TESTDIR/sample3.mp3"
+md5 -q "$MP3_TESTDIR/sample4.mp3"
+echo "pod"
+md5 -q "$POD_TESTDIR/sample1.mov"
+md5 -q "$POD_TESTDIR/sample2.mov"
+md5 -q "$POD_TESTDIR/sample3.mov"
+md5 -q "$POD_TESTDIR/sample4.mov"
+
+# delete all test files after 10 minutes
+sleep 600 && rm -r "${ROOTDIR}"
```
Comment (dericed) [inline mmtest]: add line break at end

File: mmtest [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    "${SCRIPTDIR}/makederiv" -T broadcast -F tff -s -c -o "${BR_TESTDIR}" "$file"
+    OUTPUT="${BR_TESTDIR}/$(basename "${file%.*}_SLATE.mov")"
+    echo "INGESTLOG is $INGESTLOG"
+    OUTPUT_RELATIVE_PATH=$(echo "${OUTPUT}" | sed "s:${ROOTDIR}/::g")
+    _checksumreport
+    "${SCRIPTDIR}/makederiv" -T prores -o "${PR_TESTDIR}" "$file"
+    OUTPUT="${PR_TESTDIR}/$(basename "${file%.*}.mov")"
+    OUTPUT_RELATIVE_PATH=$(echo "${OUTPUT}" | sed "s:${ROOTDIR}/::g")
+    _checksumreport
+    "${SCRIPTDIR}/makederiv" -T resourcespace -v Y -o "${RS_TESTDIR}" -d "$ROOTDIR" -e aes851@nyu.edu "$file"
```
Comment (dericed) [inline mmtest]: personal email address in the test

File: mmfunctions [hunk 4]
```diff
@@ -1441,8 +1440,13 @@ _prep_volume_adjustment(){
         if [ ! "${LOUD_ADJ}" ] ; then
             _get_volume_adjustment "${SOURCEFILE}"
         fi
-        _report -dt "Integrated loudness for $(basename "${INPUT_MOVIE}") is ${measured_I} LUFS. Reference is -24 LUFS. Will adjust by ${LOUD_ADJ} LUFS"
-        _add_audio_filter "loudnorm=measured_I=${measured_I}:measured_LRA=${measured_LRA}:measured_TP=${measured_TP}:measured_thresh=${measured_thresh}:offset=${target_offset}:linear=true:print_format=summary"
+        if [[ "${measured_I}" == "inf" ]] || [[ "${measured_I}" == "-inf" ]] ; then
+            [[ "${VOLADJUST}" == "N" ]]
```
Comment (dericed) [inline mmfunctions]: does this line do anything?

File: makeyoutube [hunk 5]
```diff
@@ -1,148 +1,17 @@
...
-    echo "  -e emailaddress ( send an email about the delivery, only valid if -d is used )"
-    echo "  -E emailaddress ( send an email about process outcome )"
-    echo "  -Y ( attempt to use uploadyoutube on the resulting file )"
-    echo "  -h ( display this help )"
     echo
-    exit
+    _report -d "makeyoutube is an alias for the updated function makederiv (options below)."
+    _report -d "You can create a file for upload to YouTube with makederiv by typing: 'makederiv -T youtube [options] [input]'."
+    _report -d "Or continue to use makeyoutube using the valid options below."
+    . "${SCRIPTDIR}/makederiv" -h
```
Comment (dericed) [inline makeyoutube]: same issue as elsewhere

File: makeresourcespace [hunk 6]
```diff
@@ -1,126 +1,17 @@
...
-    echo "  -e emailaddress ( send an email about the delivery, only valid if -d is used )"
-    echo "  -E emailaddress ( send an email about process outcome )"
-    echo "  -v Y|N (run the video through a volume adjustment filter, then adjust the volume accordingly during transcoding, slower). Defaults to ${VOLADJUST}."
-    echo "  -h ( display this help )"
     echo
-    exit
+    _report -d "makeresourcespace is an alias for the updated function makederiv (options below)."
+    _report -d "You can create a derivative file for ResourceSpace with makederiv by typing: 'makederiv -T resourcespace [options] [input]'."
+    _report -d "Or continue to use makeresourcespace using the valid options below."
+    . "${SCRIPTDIR}/makederiv" -h
```
Comment (dericed) [inline makeresourcespace]: as reported elsewhere

File: makeprores [hunk 7]
```diff
@@ -1,110 +1,17 @@
...
-    echo "  -n (dry-run mode, show the commands that would be run but don't do anything)"
-    echo "  -e emailaddress ( send an email about the delivery, only valid if -d is used )"
-    echo "  -E emailaddress ( send an email about process outcome )"
-    echo "  -h ( display this help )"
     echo
-    exit
+    _report -d "makeprores is an alias for the updated function makederiv (options below)."
+    _report -d "You can create a ProRes file with makederiv by typing: 'makederiv -T prores [options] [input]'."
+    _report -d "Or continue to use makeprores using the valid options below."
+    . "${SCRIPTDIR}/makederiv" -h
```
Comment (dericed) [inline makeprores]: !_!

File: makemp3 [hunk 8]
```diff
@@ -1,93 +1,17 @@
...
-    echo "Dependencies: ${DEPENDENCIES[@]}"
-    echo "Usage: $(basename "${0}") [ -o /directory/to/write/to/ ] fileorpackage1 [ fileorpackage2 ...]"
-    echo "  -o directory ( directory to write the resulting file to )"
-    echo "  -h ( display this help )"
     echo
-    exit
+    _report -d "makemp3 is an alias for the updated function makederiv (options below)."
+    _report -d "You can create a podcast with makederiv by typing: 'makederiv -T mp3 [options] [input]'."
+    _report -d "Or continue to use makemp3 using the valid options below."
+    . "${SCRIPTDIR}/makederiv" -h
```
Comment (dericed) [inline makemp3]: !_!

**Comment by aeschweik** [comment]
Updated with small fixes

**Pull request merged**